internal/user_settings.go

```diff
@@ -0,0 +1,55 @@
+/*
+RabbitMQ Messaging Topology Kubernetes Operator
+Copyright 2021 VMware, Inc.
+
+This product is licensed to you under the Mozilla Public License 2.0 license (the "License").  You may not use this product except in compliance with the Mozilla 2.0 License.
+
+This product may include a number of subcomponents with separate copyright notices and license terms. Your use of these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE file.
+*/
+
+package internal
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	rabbithole "github.com/michaelklishin/rabbit-hole/v2"
+	topologyv1alpha1 "github.com/rabbitmq/messaging-topology-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func GenerateUserSettings(credentials *corev1.Secret, tags []topologyv1alpha1.UserTag) (rabbithole.UserSettings, error) {
+	username, err := decodeField(credentials, "username")
```

Comment:
I would add a comment explaining that keys `username` and `password` are conforming to the provisioned service binding spec, and they are not random keys we have chosen ourselves